docs/src/userguide/extension_types.rst

```diff
@@ -76,6 +76,51 @@ and the depth attribute readable but not writable.
     Python access, not direct access. All the attributes of an extension type
     are always readable and writable by C-level access.
 
+
+.. _dynamic_attributes:
+
+Dynamic Attributes
+==================
+
+It is not possible to add attributes to an extension type at runtime by default.
+You have two ways of avoiding this limitation, both add an overhead when
+a method is called from Python code.
+
+The first workaround is making a child Python class and is preferred way of
+keeping the static attributes of an extension
+type while enabling the use of dynamic attributes::
+
+    cdef class Animal:
+
+        cdef int number_of_legs
+        def __cinit__(self, int number_of_legs):
+            self.number_of_legs = number_of_legs
+
+
+    class ExtendableAnimal(Animal):  # Note that we use class, not cdef class
+        pass
+
+
+    dog = ExtendableAnimal(4)
+    dog.has_tail = True
+
+
+Declaring a ``__dict__`` attribute is the second way of enabling dynamic attributes
+and can have a significant performance penalty compared to subclassing,
+especially when using ``cpdef`` class methods::
```

Comment:
The performance penalty really applies to both cases. It's not special to adding a `__dict__`.

Reply:
It's true, but I just wanted to say that for cpdef, the penalty is higher with the dict.